### ISSUE TITLE
before()/after() wildchar or beforeAll()/afterAll()

### ISSUE THREAD

**Issue by luke83** [deployphp/deployer#791]
| Q | A |
| --- | --- |
| Issue Type | Feature Request |
| Deployer Version | N/A |
| Local Machine OS | N/A |
| Remote Machine OS | N/A |
### Description

_I think it could be useful to have a `catch 'em all` option for after()/before() to run specific task before/after all other task. This could be useful in use cases like #777._
### Proposed interface:

```
before('*', 'timer:start');
after('*', 'timer:stop');
-or-
beforeAll('timer:start');
afterAll('timer:stop');
```

**Comment by gulios** [comment]
that's right and for parallel too because now when i do some task after('count_time') i get the longest time for server node(because task ends when all servers done) not for each server. For statistics usage it could be very useful to get information about: task_name, execution_time, server...

thx

**Comment by pluseg** [comment]
@luke83 i think that beforeAll and afterAll can lead to breaking of callback execution order. It will become less obvious. According to #777 it's easier to add profiling feature in verbose mode.

**Comment by elfet** [comment]
I think it's better to use event system here.

**Comment by elfet** [comment]
Events in v4.

**Comment by radutopala** [comment]
@elfet is the event system already implemented in 4.1?

**Comment by elfet** [comment]
No, not in v4. Probably will be in 4.1. Or maybe will back to wildchar with *.
